### PR TITLE
upgrade handling of node key constraints to v5

### LINKED ISSUE #10: Use named constraints for node keys

### LINKED ISSUE #16: Update to ITransactions to IQueryRunner
Neo4j will no longer be supporting `session.ReadTransaction` and session.WriteTransaction` as of **v6**. The methods are warned as deprecated in v5 for removal in v6.

`session.ExecuteRead` replaces `session.ReadTransaction`
`session.ExecuteWrite` replaces `session.WriteTransaction`
`IQueryRunner` as the type replaces `ITransaction` type for the **unitOfWork**

### PR DESCRIPTION
**Pull request by mdfrenchman**
- resolves #10  ; node key tests pass for v5 create, drop, exists, and exists match. Constraint is now dropped by name. node key name can be specified in the class attribute: `[Node(Label = "Car", NodeKey = "carMakeModelYear")]`
- resolves #16  ; uses IQueryRunner in preparation for v6 driver.

### CODE REVIEW

File: SchematicNeo4j/SchematicNeo4j.Tests/NodeKey/NodeKey_Drop_Tests.cs [hunk 1]
```diff
@@ -106,21 +112,19 @@ public void Dispose()
...
 
         private List<IRecord> GetConstraints(string ofType, string forLabel)
         {
             using (var session = driver.Session(o => o.WithDefaultAccessMode(AccessMode.Read)))
             {
-                return session.ReadTransaction(tx => tx.Run(
-                    "CALL db.constraints() yield description WHERE description contains (':'+$typeLabel+' ') AND description contains $constraintType RETURN description",
+                return session.ExecuteRead(tx => tx.Run(
+                    "SHOW CONSTRAINTS YIELD createStatement WHERE createStatement contains (':`'+$typeLabel+'`') AND createStatement contains $constraintType RETURN createStatement",
                     new { typeLabel = forLabel, constraintType = ofType }
```
Comment (ttemple-Calibre) [inline SchematicNeo4j/SchematicNeo4j.Tests/NodeKey/NodeKey_Drop_Tests.cs:125]: Ahhh, I think I replaced `description` with `name` in the past - will double check that and change to `createStatement`, thanks!

File: SchematicNeo4j/SchematicNeo4j/NodeKey.cs [hunk 2]
```diff
@@ -94,17 +107,19 @@ public static class NodeKey
         /// <param name="type"></param>
         /// <param name="session"></param>
         /// <returns></returns>
-        public static bool Exists(Type type, ISession session) {
-            return session.ReadTransaction(tx => Exists(type, tx));
+        public static bool Exists(Type type, ISession session)
+        {
+            return session.ExecuteWrite(tx => Exists(type, tx));
```
Comment (ttemple-Calibre) [inline SchematicNeo4j/SchematicNeo4j/NodeKey.cs:112]: Should this be an `ExecuteRead` ?

File: SchematicNeo4j/SchematicNeo4j.Tests/NodeKey/NodeKey_Drop_Tests.cs (hunk 1 above)
Comment (mdfrenchman) [reply SchematicNeo4j/SchematicNeo4j.Tests/NodeKey/NodeKey_Drop_Tests.cs:125]: @ttemple-Calibre using "createStatement" more closely mimics existing description, and ensures the labels and properties all match. By doing mutations and checks by name only, we don't have a guarantee that the constraints or indexes are equal to each other.  

If it helps, using `name` was my first thought as well. It's an easy trap to step into, and only by considering the **unhappy** edge cases does it become clear.

File: SchematicNeo4j/SchematicNeo4j/NodeKey.cs (hunk 2 above)
Comment (mdfrenchman) [reply SchematicNeo4j/SchematicNeo4j/NodeKey.cs:112]: Yes it should, will be added in next PR.

**Pull request merged**